src/Components/WebAssembly/Server/src/TargetPickerUi.cs

```diff
@@ -366,9 +366,34 @@ private string GetDevToolsUrlWithProxy(BrowserTab tabToDebug)
     {
         var underlyingV8Endpoint = new Uri(tabToDebug.WebSocketDebuggerUrl);
         var proxyEndpoint = new Uri(_debugProxyUrl);
-        var devToolsUrlAbsolute = new Uri(_browserHost + tabToDebug.DevtoolsFrontendUrl);
+        var devToolsUrlAbsolute = new Uri(new Uri(_browserHost), relativeUri: NormalizeDevtoolsFrontendUrl(tabToDebug.DevtoolsFrontendUrl));
         var devToolsUrlWithProxy = $"{devToolsUrlAbsolute.Scheme}://{devToolsUrlAbsolute.Authority}{devToolsUrlAbsolute.AbsolutePath}?{underlyingV8Endpoint.Scheme}={proxyEndpoint.Authority}{underlyingV8Endpoint.PathAndQuery}";
         return devToolsUrlWithProxy;
+
+        static string NormalizeDevtoolsFrontendUrl(string devtoolsFrontendUrl)
+        {
+            // Frontend url can be absolute or relative based on browser
+            string localPath;
+            string queryString;
+            if (Uri.TryCreate(devtoolsFrontendUrl, UriKind.Absolute, out var devtoolsFrontendUri))
+            {
+                localPath = devtoolsFrontendUri.LocalPath;
+                queryString = devtoolsFrontendUri.Query;
+            }
+            else
```

Comment:
Let's maybe skip new Uri allocations for performance reasons and stick to "else" path logic.

Splitting string into array can be replaced with 

```
int lastSlashIndex = localPath.LastIndexOf('/');
if (lastSlashIndex >= 0)
{
    localPath = localPath.Substring(lastSlashIndex + 1);
 }
```

Reply:
I used Uri as "the correct way" to parse url because of what javier said.. otherwise other parts of code in this and another file can be improved for performance!
However the "else" is a fallback for who (I don't know if it is possible) use an outdated version of the browser which cannot be updated or if for (some reason) the urls will change again in the future return back to relative kind.